### DISCUSSION TITLE
Equivalent to a SQL `LIMIT` or `GROUP BY` in the client SDK?

### DISCUSSION THREAD

**Discussion by Aragur** [pocketbase/pocketbase#1918] [Q&A] [answered]
Currently, I'm stuck implementing a special case in the frontend.
I'm creating a chat functionality using pocketbase as the backend.
I saved my messages with a schema containing the author, the recipient, and the message.
Now I want to order all the chat partners by when the latest message was sent.
Currently, the only way is to get a full list, filtered by the userId, and then manually via a for each loop creating this list.
I'm missing a ```LIMIT``` Operator or a ```GROUP BY``` which is supported by SQL.
Am I missing something here, or is this currently not possible using the SDK?

**Comment by ganigeorgiev** [comment]
> I saved my messages with a schema containing the author, the recipient, and the message.
> Now I want to order all the chat partners by when the latest message was sent.

I don't understand the structure. How the "chat partners" are related to the messages? For what do you need `GROUP BY`?


In any case, via the web APIs currently you can apply `LIMIT` to a search result using the `perPage` argument:

```js
const result = await pb.collection("example").getList(1, 100);
```

Client-side `GROUP BY` or aggregations in general are not supported via the web APIs and there are no plans for that since it will be hard to define the security rules or to express an aggregation with just the filters.

In the upcoming v0.13.0 release there will be support for a "View" collection type that will allow creating a readonly collection from any SQL `SELECT` query.

**Reply by Aragur** [reply]
This was perfectly fixed by using the new `collection view` on the latest v0.13.0 release.
Thanks a lot :)

**Comment by Aragur** [comment, marked as answer]
This was perfectly fixed by using the new collection view on the latest v0.13.0 release.
Thanks a lot :)
